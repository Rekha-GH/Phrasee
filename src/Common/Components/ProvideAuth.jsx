import React, { createContext } from 'react';
import { useSelector } from 'react-redux';
import { getSession } from '../../Login/Store/LoginSelector';

const authContext = createContext();
const ProvideAuth = React.memo(({ children }) => {
    const auth = useSelector(getSession);

    return (
        <authContext.Provider value={auth}>
            {children}
        </authContext.Provider>
    );
});

ProvideAuth.displayName = 'ProvideAuth';
export default ProvideAuth;