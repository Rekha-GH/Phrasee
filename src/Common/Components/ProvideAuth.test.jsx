import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import ProvideAuth from './ProvideAuth';
import { shallow } from '../../setupTests';

describe('ProvideAuth Tests', () => {
  const useSelectorMock = jest.spyOn(ReactRedux, 'useSelector');

  test('ProvideAuth renders correctly', () => {
    useSelectorMock.mockImplementationOnce(() => { session: 'test' }); 
    const wrapper = shallow(<ProvideAuth><div>test</div></ProvideAuth>).dive();
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
