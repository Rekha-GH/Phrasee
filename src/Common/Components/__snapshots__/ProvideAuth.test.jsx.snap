// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProvideAuth Tests ProvideAuth renders correctly 1`] = `
<div>
  test
</div>
`;
